### DISCUSSION TITLE
Is it possible to "back-expand" relations in an API rule?

### DISCUSSION THREAD

**Discussion by parkernilson** [pocketbase/pocketbase#3515] [Q&A] [answered]
I have a 'users' table with a multiple-relation to 'lists', and 'lists' has a multiple relation to 'list_items'.

In my API rules, I would like to only let a user view 'list_items' if they own the list that the list items belong to.

Am I able to write an api rule that will 'back-expand' the relation from 'list_items' so that I can check the 'owner_id' on the list that the list item belongs to?

**Comment by ganigeorgiev** [comment, marked as answer]
Back-relation filtering and sorting currently is not supported (_it is in the roadmap and will be implemented eventually in the near future_).

As a workaround, the API rules currently allow performing a "join" using the `@collection.*` rule. There are not enough details about the names of your fields, but assuming that the relation fields are named the same way as their related collections then I guess you can try with a "list_items" API rule like:

```js
@request.auth.id != "" &&
@collection.lists.list_items.id ?= id &&
@collection.users.lists.id ?= @collection.lists.id &&
@collection.users.id ?= @request.auth.id

// (edit) OR directly through users
@request.auth.id != "" &&
@collection.users.id ?= @request.auth.id &&
@collection.users.lists.list_items.id ?= id
```
(I haven't tested it but the above should allow list items access to the authenticated "users" that have _at least one_ "lists" containing the list item in question)

_Alternatively, if you use a more traditional SQL approach and reverse the relation, aka. move the `relation` to the child table and/or use junction collections that will contain the many-many single relations, then you should be able to traverse the relation tree using dot-notation._

**Reply by parkernilson** [reply]
Perfect! I will try it out in a bit. That is very helpful!

**Reply by ganigeorgiev** [reply]
_edit: I've update the rule since I realized that you can skip the explicit `@collection.lists.*` join and perform it through the `@collection.users.lists`._
